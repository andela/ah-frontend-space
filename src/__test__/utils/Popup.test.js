import React from 'react';
import { shallow } from 'enzyme';
import {
  Popup,
} from '../../utils/PopUp';

describe('render PopUp with shallow', () => {
  const slug = 'slug';
  const defaultProps = {
    slug,
    closePopup: jest.fn(),
    actions: {
      deleteArticleAction: () => new Promise((resolve) => {
        resolve('Success Message');
      }),
    },
    loading: false,
  };

  const wrapper = shallow(<Popup {...defaultProps} />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should perform onClick to delete article', () => {
    const wrapperInstance = wrapper.instance();
    wrapperInstance.deleteArticle();
  });
});
