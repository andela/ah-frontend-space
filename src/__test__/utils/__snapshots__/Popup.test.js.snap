// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PopUp with shallow should match snapshot 1`] = `ShallowWrapper {}`;
